### DISCUSSION TITLE
Display foreign Website in my Website

### DISCUSSION THREAD

**Discussion by antonius774** [unoplatform/uno#11814] [Q&A]
We would like to port our prototype Apps to UNO and WebAssembly to have a single code base for all platforms and also enable people to use us from the browser. 
Unfortunately we use WebViews and display foreign websites in the native Apps. And that triggers the Single Origin Policy when we just try to port it to the web. I was hoping to get around that with WebAssembly cause I was told it is possible to go around SOP this way. Sounded credible to me cause Wasm seems  to run so deep/ native like. 
But since then I am super struggling to actually find a way to implement it. 
I was hoping WebViews in Uno could help me with that. Sadly currently WebViews do not seem to be supported in uno and probably wouldn't bypass the SOP anyways.

Does anyone has any other ideas how one could achieve that? Or do you know some good documentation where I could better understand how wasm is actually executed and how web connectivity is implemented and filtered?

**Comment by jeromelaban** [comment]
Thanks for the question. Indeed, bypassing SOP is not a good practice, even when possible.

While WebView2 for WebAssembly is not implemented (https://github.com/unoplatform/uno/issues/329), you can create your own control that wraps an IFRAME, as discussed in https://github.com/unoplatform/uno/discussions/7739. You may also need to have the origin site adhere to compatible CORS headers for this solution to work.

**Reply by antonius774** [reply]
Thanks you for you swift answer, I apologize this is all very new to me so it takes me some time to go thru your suggestions.

What I didn't understand following #7739 I would wrap my iframe in JS. How is that meant to bypass the SOP. Wouldn't embedding JS Components into my UNO-Wasm give me the same limitations I had with JS in the first place?

Also do you by any chance know a display/graph or doc page which would help me to understand how internet/network connectivity is interfaced between my wasm and the browser? I am still very unsure how communicating and loading stuff into the sandbox is achieved.

**Reply by jeromelaban** [reply]
WebAssembly in the discussion is not relevant, as it cannot do what JS cannot do. If you need help specifically on what iframe can do, you'll find better information on stackoverflow or other browser specific documentation.

The suggestion about wrapping an iframe for Uno Platform is only about being able to integrate an iframe in an Uno Platform App, not about working around any kind of browser restrictions.
